internal/backend/connections/connections.go

```diff
@@ -54,9 +121,30 @@ func (c *Connections) Get(ctx context.Context, id sdktypes.ConnectionID) (sdktyp
 		return sdktypes.InvalidConnection, err
 	}
 
-	return desc, nil
+	return c.enrichConnection(ctx, desc)
 }
 
-func (c *Connections) List(ctx context.Context, filter sdkservices.ListConnectionsFilter) ([]sdktypes.Connection, error) {
-	return c.DB.ListConnections(ctx, filter, false)
+func (c *Connections) enrichConnection(ctx context.Context, conn sdktypes.Connection) (sdktypes.Connection, error) {
+	intg, err := c.Integrations.GetByID(ctx, conn.IntegrationID())
```

Comment:
future update, but we should definitely cache this integration objects or perform one db call to fetch connection with the integration
otherwise list connections would double the db call (get integration for each connection so twice the calls)

Reply:
welp currently it just looks in a map since integrations are not backed up by a db, but i agree.